difforig perl-5.10.1/lib/Cwd.pm

by Eliot Moss 2009

When running svk, a package written in perl that runs
"on top of" svn, I have found the attached patch to be
necessary. Otherwise svk does not handle filenames
properly. The first change is the essential one. The
second one just eliminates an annoying and possibly
confusing/alarming (but I think truly harmless)
warning about setting something a second time. (The
affected lines were only recently introduced, I
believe.) This has been going on for some time, and
I finally got organized to send this in, rather than
continuing to edit the file myself after each update. 


diff -u perl-5.10.1/lib/Cwd.pm.orig perl-5.10.1/lib/Cwd.pm
--- perl-5.10.1/lib/Cwd.pm.orig	2009-07-06 10:18:42.000000000 +0000
+++ perl-5.10.1/lib/Cwd.pm	2009-12-18 14:45:54.942875000 +0000
@@ -305,7 +305,7 @@
     getcwd		=> 'cwd',
     fastgetcwd		=> 'cwd',
     fastcwd		=> 'cwd',
-    abs_path		=> 'fast_abs_path',
+#   abs_path		=> 'fast_abs_path',
     realpath		=> 'fast_abs_path',
    },
 
@@ -819,6 +819,6 @@
 
 # added function alias for those of us more
 # used to the libc function.  --tchrist 27-Jan-00
-*realpath = \&abs_path;
+*realpath = \&abs_path unless defined &realpath;
 
 1;
